packages/ts-transformers/src/internal/decorators/query-assigned-elements.ts

```diff
@@ -251,4 +264,26 @@ export class QueryAssignedElementsVisitor implements MemberDecoratorVisitor {
       getterBody
     );
   }
+
+  /**
+   * @param slot Expression that evaluates to the slot name.
+   * @returns Template string node representing `slot[name=${slot}]`
+   */
+  private createNamedSlotSelector(slot: ts.Expression): ts.TemplateExpression {
+    const factory = this._factory;
+    return factory.createTemplateExpression(
```

Comment:
Consider checking if slot is a string literal here, and generating a pre-interpolated string here (e.g. `"slot[name=foo]"` instead of `"slot[name=${"foo"}]`). That will be faster (no need for a runtime interpolation), and also more like what a human would usually work (important because the main use case of the transformers is actually docs generation).

Reply:
Genius comment. Thank you. Added the special handling for string literals and no substitution template literals. I.e. "foo" or `foo`. Also updated tests so both are tested.